lxd/main_daemon.go

```diff
@@ -80,19 +80,23 @@ func (c *cmdDaemon) Run(cmd *cobra.Command, args []string) error {
 	select {
 	case sig := <-ch:
 		if sig == unix.SIGPWR {
-			logger.Infof("Received '%s signal', shutting down containers", sig)
+			logger.Infof("Received '%s signal', shutting down instances", sig)
 			d.Kill()
-			containersShutdown(s)
+			instancesShutdown(s)
 			networkShutdown(s)
 		} else {
 			logger.Infof("Received '%s signal', exiting", sig)
 			d.Kill()
 		}
 
 	case <-d.shutdownChan:
```

Comment:
@monstermunchkin this doesn't look quite right, the "shutdown channel" is effectively being replaced with the global context, so when its cancelled this indicates a shutdown should occur. However here the code is waiting for something from the shutdown channel and then passing it into `waitForOperations` which also internally waits on it. Also it would be clearer to call it forceShutdownChan to indicate it is different from the normal shutdown process.

Reply:
@tomponline Are you saying we should get rid of the shutdown channel altogether?

Reply:
@monstermunchkin yes that was my idea, you can wait on `case <-ctx.Done():` from the context to indicate that shutdown has been requested.

Reply:
You can use `case <-ctx.Done()` but that would always lead to a forceful shutdown in our case. Example:
```go
func main() {
	ctx, cancel := context.WithCancel(context.Background())
	
	go func(cancel context.CancelFunc) {
		time.Sleep(1)
		cancel()
	}(cancel)
	
	// this will block until cancel() is called
	<-ctx.Done()
	// this (unfortunately) won't block again
	<-ctx.Done()
}
```

That means, if we replaced the shutdown channel with `case <- ctx.Done()` in `waitForOperations()`, it would always take that code path.